CONTRIBUTIONS.md

```diff
@@ -54,33 +54,40 @@ A lot of people are excited about building a JSON API in Servant and everyone se
 
 ## UI languages/frameworks
 
-| Library/framewor | Who's working on it |
-| --- | --- |
-| Elm | No one yet |
-| Purescript | No one yet |
-| Reflex FRP | No one yet |
-
+| Library/framework | Who's working on it                     |
+| ---               | ---                                     |
+| Elm               | No one yet                              |
+| Purescript        | No one yet                              |
+| Reflex FRP        | [meditans](https://github.com/meditans) |
 
 # Higher order design goals
 
+## General principles
+* Making nonsensical states non-representible in the domain model. eg. product can have only two type -- physical and digital. Status can have only few values. Basically a lot of ADT usage.
+* Lifting more invariants to the type-level
+* Unit tests
+* Controller tests
+* Integration tests
+
+## Database domain
+
 * Best way to deal with housekeeping columns, like `createdAt`, `updatedAt`
 * Implementing audit logs
 * Ensuring type-safety in the create/edit/update calls for each DB model, such that an accidental write of a "protected" field is not possible. Example of such fields: `createdAt`, `updatedAt`, `id`, `status`, `type`, etc. (fields that need some side-effects or workflow to change)
 * How to deal with DB updates? 
   * Should the domain API take the complete record as an argument? Who should be responsible for loading the record from the DB? How many times will we be loading the same record from the DB, if we need to chain/compose different update APIs together?
   * Should the domain API take a *diff* as an argument? How do we represent a diff in a typesafe manner?
-* Making nonsensical states non-representible in the domain model. eg. product can have only two type -- physical and digital. Status can have only few values. Basically a lot of ADT usage
 * Implementing validations
-* Lifting more invariants to the type-level
 * Changing response JSON based on incoming request
 * DB transactions
 * Implementing authorization
 * JSONB, ENUM, and Array support in DB library
 * Redis caching at object level
 * Redis caching at page level
-* Unit tests
-* Controller tests
-* Integration tests
+
+## UI Domain
+* Creating a reusable collection of abstractions for most used ui components (forms etc.)
+* Investigate the tradeoff of doing all the implementation in haskell vs. interface, via haskell, to html templates.
```

Comment:
More stuff to think about:

* Integration with existing jQuery widgets (calendar, accordion, search/sort tables, editable grids, etc)
* Minification toolchain to reduce final JS size
* Progressive loading of JS files to reduce initial page-load time
* Server-side rendering of initial page-load
* Sharing data structures with backend
* "Adaptive validation" - it should be possible to quickly throw together an *acceptable* form, which depends on the backend for data validation. The backend should send error messages in a structured format so that the form is able to *reasonably* place the error messages next to the field (or fieldset) with the bad data. If the form's UX is very important, then validations can be implemented on the JS side as well, but the idea is that it **should be possible** to have a working form without client-side validation.